src/Sulu/Bundle/PreviewBundle/Preview/PreviewCache.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\PreviewBundle\Preview;
+
+use Doctrine\Common\Cache\Cache;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * @internal BC Layer between doctrine and symfony cache
+ */
+class PreviewCache
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cache;
+
+    /**
+     * @param CacheItemPoolInterface|Cache $cache
+     */
+    public function __construct($cache)
+    {
+        if ($cache instanceof Cache) {
+            @trigger_error(
+                sprintf(
+                    'To inject $cache as instance of "%s" is deprecated use a "%s" instead.',
+                    get_class($cache),
+                    CacheItemPoolInterface::class
+                ),
+                E_USER_DEPRECATED
+            );
+        }
+
+        if (!$cache instanceof Cache && !$cache instanceof CacheItemPoolInterface) {
+            throw new \RuntimeException(
+                sprintf(
+                    'The $cache need to be an instance of "%s" or "%s" but got "%s".',
+                    CacheItemPoolInterface::class,
+                    Cache::class,
+                    get_class($cache)
+                )
+            );
+        }
+
+        $this->cache = $cache;
+    }
+
+    public function delete(string $id): void
+    {
+        $this->cache->delete($id);
+    }
+
+    public function contains(string $id): bool
```

Comment:
Would it be easily possible to use the naming of the new `Cache` component here? Then removing this layer will be easier later.

Reply:
Or do you think this class will stay anyway? I've seen that some methods consist of more than just a single call.

Reply:
I personally would keep it but stay as internal class.